src/main/java/htsjdk/samtools/cram/build/CRAMReferenceRegion.java

```diff
@@ -113,9 +113,14 @@ public void fetchReferenceBases(final int referenceIndex) {
 
         // Re-resolve the reference bases if we don't have a current region or if the region we have
         // doesn't span the *entire* contig requested.
+        final SAMSequenceRecord newSequenceRecord = sequenceDictionary.getSequence(referenceIndex);
+        if (newSequenceRecord == null) {
+            throw new IllegalArgumentException(
+                    String.format("Requested reference sequence index %d not found", referenceIndex));
+        }
         if ((referenceIndex != this.referenceIndex) ||
                 regionStart != 0 ||
-                (regionLength < referenceBases.length)) {
+                (regionLength != newSequenceRecord.getSequenceLength())) {
             setCurrentSequence(referenceIndex);
```

Comment:
Add a comment that the `setCurrentSequence()` call needs to happen first in this `if` block (in particular, before the assignment to `regionLength` below). Or alternatively, you could have `regionLength = newSequenceRecord.getSequenceLength();` below to eliminate the dependency on order of operations.

Reply:
Latter is done.